### ISSUE TITLE
Exponential Websocket Connection Errors

### ISSUE THREAD

**Issue by Nolence** [giantmachines/redux-websocket#53]
In my app I've setup the store to look like this:

```javascript
const rootReducer = combineReducers({
  /* ... */
  ws: wsReducer
});

const reduxWebsocketMiddleware = reduxWebsocket({ reconnectInterval: 5000 });

let store = createStore(
    rootReducer,
    applyMiddleware(reduxWebsocketMiddleware)
  );
```

In another file, I try to setup the connection:

```javascript
useEffect(() => {
  wsConnect();
  return () => wsDisconnect();
}, []);

/* ... */

const mapDispatchToProps = dispatch => ({
  wsConnect: () => dispatch(connect("ws://127.0.0.1:5678/")),
  wsDisconnect: () => dispatch(disconnect()),
});
```

It successfully tries to connect. However, less than thirty seconds later, it's sending about 1000 requests per second, freezing my tab up. the `useEffect` is only being called once. My wsReducer is letting the state fall straight through. Not sure what is causing this but previous to `1.0.0`, this same code worked.

![redux-websocket-errors](https://user-images.githubusercontent.com/29642168/56482357-004b3b80-6481-11e9-95a8-15e5883a2170.PNG)

**Comment by brianmcallister** [comment]
Hey @Nolence, thanks for providing all of this info. Is it possible that the connection action is getting dispatched multiple times? Each time you dispatch a connection action, a new connection is created (and that includes another WebSocket instance), so I think it might be the case that you're unintentionally creating many WebSocket instances.

**Comment by Nolence** [comment]
Thanks for helping me out @brianmcallister. I'm really not sure though if I'm calling it multiple times. I looked in several places but couldn't find anything.

Here's a very minimal example of the error I'm talking about: https://github.com/Nolence/redux_websocket_errors

**Comment by brianmcallister** [comment]
This is very helpful, thanks.

I cloned that repo, and noticed a few things. When I switched the server URL to `wss://websocket-echo-server.herokuapp.com`, everything worked fine. I only saw the errors and behavior you described when trying to connect to `ws://127.0.0.1:5678/`. In that case, I immediately saw a flood of errors because I did not have a server running at that address.

This leads me to believe that the problem is within `redux-websocket`, and how we're not correctly handling connection errors.

I'll make a change that handles this error and publish a new minor version as soon as I'm able. For now, you might want to stick with the previous version until I can get this fix published.

Thanks again for the test case, that was very helpful.
